fuseparts/_fusemodule.c

```diff
@@ -1272,15 +1272,31 @@ poll_func(const char *path, struct fuse_file_info *fi,
 	  struct fuse_pollhandle *ph, unsigned *reventsp)
 {
 	PyObject *pollhandle = Py_None;
+	int ret = -EINVAL;
+	PyObject *v;
+
+	PYLOCK();
 
 	if (ph)
 		pollhandle = PyCapsule_New(ph, pollhandle_name, pollhandle_destructor);
```

Comment:
On the note of possible existing errors - if we're returning `Py_None`, I think its refcount must be increased or we risk underflowing it. Can you confirm?

Reply:
> There is actually another exiting bug in the code. `PyCapsule_New()` could return with error, so we need to check that too.

Should we really call `PyErr_Print()`? I'd expect `PyCapsule_New()` to set an exception?

Reply:
> if we're returning Py_None, I think its refcount must be increased or we risk underflowing it. 

I think `Py_None` works without refcount because of `if (ph) Py_DECREF(pollhandle);`. We aren't actually returning it, but rather passing it as an arg to another function.

> Should we really call PyErr_Print()? I'd expect PyCapsule_New() to set an exception?

Yes, it does set an exception. But since this is a callback into libfuse we can't leave the exception set. Libfuse doesn't know how to handle that and Python may never run on this thread again to handle the exception later. The best we can can do is print the unhandled exception and return negative error code to libfuse.